packages/react/src/ReactBaseClasses.js

```diff
@@ -117,44 +117,56 @@ if (__DEV__) {
   }
 }
 
-/**
- * Base class helpers for the updating state of a component.
- */
+function ComponentDummy() {}
+ComponentDummy.prototype = Component.prototype;
+
+function configurePrototype(ComponentSubclass, prototypeProperties) {
+  const prototype = (ComponentSubclass.prototype = new ComponentDummy());
+  prototype.constructor = ComponentSubclass;
+
+  // Avoid an extra prototype jump for these methods.
+  Object.assign(prototype, Component.prototype);
+
+  // Mixin additional properties
+  Object.assign(prototype, prototypeProperties);
+}
+
+// Convenience component with default shallow equality check for sCU.
 function PureComponent(props, context, updater) {
-  // Duplicated from Component.
   this.props = props;
   this.context = context;
   this.refs = emptyObject;
-  // We initialize the default updater but the real one gets injected by the
-  // renderer.
   this.updater = updater || ReactNoopUpdateQueue;
 }
+configurePrototype(PureComponent, {isPureReactComponent: true});
 
-function ComponentDummy() {}
-ComponentDummy.prototype = Component.prototype;
-const pureComponentPrototype = (PureComponent.prototype = new ComponentDummy());
-pureComponentPrototype.constructor = PureComponent;
-// Avoid an extra prototype jump for these methods.
-Object.assign(pureComponentPrototype, Component.prototype);
```

Comment:
Are we sure this matters these days in any significant way? Maybe we can just kill this.